### PR TITLE
[ZT-385] buildNpmPackage: build node_modules in a separate derivation

### PR DESCRIPTION
**Pull request by zhenyavinogradov**
Avoids rebuilding `node_modules` if package.json didn't change.

`npmPackPhase` moved into `installPhase` to make it easier to override

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -110,50 +110,73 @@ with stdenv.lib; let
...
       installPhase = ''
         runHook preInstall
-        ${untarAndWrap info.name [npmCmd]}
+        npm prune --production
+        npm pack --ignore-scripts
+        ${untarAndWrap name [npmCmd]}
         runHook postInstall
       '';
-    } // commonEnv // args // {
+    } // commonEnv // extraEnvVars // removeAttrs args [ "extraEnvVars" ] // {
```
Comment (yorickvP) [inline default.nix]: why have extraEnvVars if you could just add things to `args`?

**Comment by zhenyavinogradov** [comment]
`extraEnvVars` in for variables you want to pass to `mkNodeModules` as well

**Comment by yorickvP** [comment]
Okay, could you look at the failing builds?

**Comment by zhenyavinogradov** [comment]
The CI is just building a derivation from someone's abandoned github repo using `buildYarnPackage`. I suggest to remove the CI altogether, and do tests in the repositories where we use this library

**Comment by yorickvP** [comment]
Having CI makes it significantly easier to review code changes like this one, though. Maybe we can write a simple testcase.

**Comment by zhenyavinogradov** [comment]
I've added a simple test. Also apparently this library doesn't work with current nixpkgs (unrelated to this PR), I'll look into that

**Pull request merged**